src/components/formElements/ToggleSwitch.vue

```diff
@@ -0,0 +1,134 @@
+<script>
+export default {
+  props: {
+    isChecked: {
+      type: Boolean,
+      default: false,
+    },
+    stopPropagation: {
+      type: Boolean,
+      default: false,
+    },
+    title: String,
+  },
+  emits: ['change'],
+};
+</script>
+
+<template>
+  <label
+    class="toggle-switch"
+    :class="[
+      { checked: isChecked },
+      `${globalTheme}-theme`
+    ]"
+    @change="$emit('change')"
+  >
+    <div class="title"> {{ title }}</div>
+    <input
+      class="input"
+      type="checkbox"
+      :stop-propagation="stopPropagation"
+      :checked="isChecked"
+    >
+    <span class="status" />
+  </label>
+</template>
+
+<style lang="scss" scoped>
+.toggle-switch {
+  display: block;
+  position: relative;
+  height: 20px;
+  color: map-get($colors, 'black');
+  background-color: map-get($colors, 'gray-very-light');
+  border: 1px solid map-get($colors, 'gray-light');
+  border-radius: 5px;
+  cursor: pointer;
+  transition: 0.3s;
+
+  &:hover {
+    background-color: map-get($colors, 'white');
+    border: 1px solid map-get($colors, 'gray-dark');
+  }
+
+  .title {
+    padding: 0 32px 0 5px;
+    white-space: nowrap;
+    font-size: 12px;
+    font-variant: small-caps;
+  }
+
+  .status {
+    &:before {
+      content: "off";
```

Comment:
it's better to do with html
because of that add translate in scss is gonna be almost impossible